### ISSUE TITLE
Final Report Peer Review (gs484)

### ISSUE THREAD

**Issue by genghisshyy** [jhc0825/Stock-Prediction-Using-Social-Media-Data#14]
This project explores the ability of social media to predict stock performance, specifically for five of the largest companies in the tech industry today: Facebook, Apple, Amazon, Netflix, and Google. Posts and tweets about these five companies were scraped from Reddit and Twitter, respectively, with all collected data being originally posted in the timeframe from the beginning of July 2020 to the end of September 2020. Similarly, stock performance statistics for these five companies during this same timeframe were downloaded from Finnhub, allowing the group to achieve their ultimate goal: to either develop some strategy of accurately predicting stocks based on social media activity, or investigate why such a strategy is not plausible.

To begin with, this group certainly deserves recognition in regard to their data collection process, as they were able to overcome various API restrictions and cleanly eliminate a significant amount of non-relevant data in order to construct their final dataset. (Granted, this dataset could have been expanded to cover a larger timeframe, which probably would have improved results; but the time constraints hindering the creation of such a dataset were at least clearly noted in the report.) Beyond data collection, this group also effectively backed their conclusions with a variety of data visualizations, such as the line graphs depicting misclassification error over number of lag days for the hinge loss model and the scatter plots used to illustrate the lack of strong correlation between tweet activity about a given company and that company's stock market returns. Similarly, the group's decision to first test each of their models on all five companies individually before testing each of their models on all five companies together served as a thorough analysis of how social media might yield more or less accurate predictions for stock performance depending on the company in question.

However, the group's justifications for some of their major project decisions were not so well laid out. For example, when explaining their decision to split data chronologically into a training set (containing data from August 2020) and a test set (containing data from September 2020), no strong explanation is provided, aside from stating this method made sense to them "as stock investors personally." Even if this strategy were to make sense, a better explanation would certainly have been helpful to the audience (especially readers who do not have experience in stock investment.) In addition, the group concludes by stating their final model achieved their "initial goal of...[creating] a model with a mean error rate of less than 50%." This goal was not clearly articulated in the introduction of the report, and was indeed not even mentioned during the group's discussion of their linear regression model and initial classification model using logistic loss. Stating this goal at the start of the project, along with a justification for why the "threshold" mean error rate was set at 50%, would again have greatly eased the audience's understanding of the overall study. Finally, although brief explanations were given for why hinge and logistic loss were utilized, no explanation was given regarding the choice of regularizers (specifically, a quadratic regularizer.) Not indicating why quadratic regularization was chosen over other forms of regularization arguably leaves the audience wondering  whether other regularizers would have improved the models that were eventually developed.
